runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -303,22 +331,33 @@ public void onMenuOptionClicked(MenuOptionClicked click)
 			return;
 		}
 
-		if (removed)
+		if (click.getMenuOption().equals(TAG) || click.getMenuOption().equals(UNTAG))
 		{
-			highlightedNpcs.remove(npc);
-			memorizedNpcs.remove(npc.getIndex());
-		}
-		else
-		{
-			if (!client.isInInstancedRegion())
+			final boolean removed = npcTags.remove(id);
+
+			if (removed)
 			{
-				memorizeNpc(npc);
-				npcTags.add(id);
+				highlightedNpcs.remove(npc);
+				memorizedNpcs.remove(npc.getIndex());
 			}
-			highlightedNpcs.add(npc);
+			else
+			{
+				if (!client.isInInstancedRegion())
+				{
+					memorizeNpc(npc);
+					npcTags.add(id);
+				}
+				highlightedNpcs.add(npc);
+			}
+
+			click.consume();
 		}
+		else
+		{
+			final String name = npc.getName().toLowerCase();
```

Comment:
This lowercase isn't necessary and just causes it to be added to the user's list in lowercase.